### ISSUE TITLE
NYTimes videos fail with "503: Backend is unhealthy" errors

### ISSUE THREAD

**Issue by rsfinlayson** [yt-dlp/yt-dlp#8605]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

new zealand

### Provide a description that is worded well enough to be understood

NYTimes videos play OK in browsers (Firefox, Safari, Chrome), but when I try to download them with yt-dlp, there is a "503: Backend is unhealthy" error.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://nyti.ms/47gEoFW']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.11.16 from yt-dlp/yt-dlp [24f827875] (pip)
[debug] Python 3.12.0 (CPython x86_64 64bit) - macOS-14.1.1-x86_64-i386-64bit (OpenSSL 3.1.4 24 Oct 2023)
[debug] exe versions: ffmpeg 3.3.4, ffprobe 3.3.4, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.19.0, brotli-1.1.0, certifi-2023.07.22, mutagen-1.47.0, requests-2.31.0, sqlite3-3.44.0, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests
[debug] Loaded 1901 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: stable@2023.11.16 from yt-dlp/yt-dlp
yt-dlp is up to date (stable@2023.11.16 from yt-dlp/yt-dlp)
[generic] Extracting URL: https://nyti.ms/47gEoFW
[generic] 47gEoFW: Downloading webpage
[redirect] Following redirect to https://www.nytimes.com/video/us/politics/100000009183577/biden-xi-apec.html?smid=pl-share
[NYTimes] Extracting URL: https://www.nytimes.com/video/us/politics/100000009183577/biden-xi-apec.html?smid=pl-share
[NYTimes] 100000009183577: Downloading video JSON
WARNING: [NYTimes] Unable to download JSON metadata: HTTP Error 503: Backend is unhealthy
[NYTimes] 100000009183577: Downloading video JSON
ERROR: [NYTimes] 100000009183577: Unable to download JSON metadata: HTTP Error 503: Backend is unhealthy (caused by <HTTPError 503: Backend is unhealthy>); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 715, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/nytimes.py", line 128, in _real_extract
    return self._extract_video_from_id(video_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/nytimes.py", line 30, in _extract_video_from_id
    video_data = self._download_json(
                 ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 1069, in download_content
    res = getattr(self, download_handle.__name__)(url_or_request, video_id, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 1033, in download_handle
    res = self._download_webpage_handle(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 903, in _download_webpage_handle
    urlh = self._request_webpage(url_or_request, video_id, note, errnote, fatal, data=data, headers=headers, query=query, expected_status=expected_status)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 860, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/YoutubeDL.py", line 4052, in urlopen
    return self._request_director.send(req)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/networking/common.py", line 114, in send
    response = handler.send(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/networking/_helper.py", line 204, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/networking/common.py", line 325, in send
    return self._send(request)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/networking/_requests.py", line 341, in _send
    raise HTTPError(res, redirect_loop=max_redirects_exceeded)
yt_dlp.networking.exceptions.HTTPError: HTTP Error 503: Backend is unhealthy

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/extractor/common.py", line 847, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/Cellar/yt-dlp/2023.11.16/libexec/lib/python3.12/site-packages/yt_dlp/YoutubeDL.py", line 4074, in urlopen
    raise _CompatHTTPError(e) from e
yt_dlp.networking.exceptions._CompatHTTPError: HTTP Error 503: Backend is unhealthy
```

**Comment by SirElderling** [comment]
All tests for this extractor are failing. 
My assumption is that NYTimes changed at some point the way videos are served.

There appears to be an `application/ld+json` script embedded in page that contains most of the information required for extraction

```json
{
  "@context": "http://schema.org",
  "@type": "VideoObject",
  "@id": "https://www.nytimes.com/video/us/politics/100000009183577/biden-xi-apec.html",
  "description": "President Biden said that China’s president, Xi Jinping, had agreed to resume military-to-military communication.",
  "url": "https://www.nytimes.com/video/us/politics/100000009183577/biden-xi-apec.html",
  "name": "Biden, Xi Make Progress on Military Talks and Fentanyl",
  "thumbnailUrl": [
    "https://static01.nyt.com/images/2023/11/15/multimedia/15dc-prexy-zbqk/15dc-prexy-zbqk-videoSixteenByNineJumbo1600-v2.jpg",
    "https://static01.nyt.com/images/2023/11/15/multimedia/15dc-prexy-zbqk/15dc-prexy-zbqk-superJumbo-v2.jpg",
    "https://static01.nyt.com/images/2023/11/15/multimedia/15dc-prexy-zbqk/15dc-prexy-zbqk-mediumSquareAt3X-v2.jpg"
  ],
  "publisher": {
    "@context": "http://schema.org",
    "@type": "Organization",
    "name": "The New York Times",
    "logo": {
      "@context": "http://schema.org",
      "@type": "ImageObject",
      "url": "https://static01.nyt.com/images/misc/NYT_logo_rss_250x40.png",
      "height": 40,
      "width": 250,
      "contentUrl": "https://static01.nyt.com/images/misc/NYT_logo_rss_250x40.png",
      "creditText": "The New York Times"
    },
    "url": "https://www.nytimes.com/"
  },
  "embedUrl": "https://www.nytimes.com/video/players/offsite/index.html?videoId=100000009183577",
  "contentUrl": "https://vp.nyt.com/video/2023/11/16/113401_1_16vid-biden-apec_wg_240p.mp4",
  "uploadDate": "2023-11-16T02:46:33.000Z",
  "transcript": "We’re reassuming military-to-military contact, direct contacts. As a lot of you press know or follow this, that’s been cut off. It’s been worrisome. That’s how accidents happen – misunderstandings. So we’re back to direct, open, clear, direct communications on a direct basis. Vital miscalculations on either side can cause real, real trouble with a country like China or any other major country. And so I think we’re making real progress there as well. Do I trust? I trust but verify, as the old saying goes. That’s where I am. And, you know, we’re in a competitive relationship - China and the United States. But my responsibility is to make this rational and manageable, so it - so it doesn’t result in conflict. That’s what I’m all about. That’s what this is about. We’re in a situation where we agreed that fentanyl and its precursors will be curbed substantially – and the pill presses. That’s a big moment.",
  "duration": "PT1M12.172S"
}
```
I may be wrong, but after some testing I think the following pattern may be used to extract the m3u8 for a given video: 

```
https://vp.nyt.com/video/hls/<year>/<month>/<date>/<video_file_name_without_quality>/master.m3u8
```


Examples:

```
URL: https://www.nytimes.com/video/us/politics/100000009183577/biden-xi-apec.html
m3u8: https://vp.nyt.com/video/hls/2023/11/16/113401_1_16vid-biden-apec_wg/master.m3u8

URL: https://www.nytimes.com/video/us/100000006525294/ufo-video-navy.html
m3u8: https://vp.nyt.com/video/hls/2019/05/24/80888_1_24vid-UFO_wg/master.m3u8
```

**Comment by dirkf** [comment]
The JSON API URL used by the NYT extractors to resolve the video ("media asset") is apparently obsolete, both v3 and v2 (tried if that fails).

The turkey roast page loads https://static01.nyt.com/video-static/vhs3/vhs.min.js and then posts the request that appears in this _curl_ command, according to Mozilla devtools.
```
curl 'https://samizdat-graphql.nytimes.com/graphql/v2' -H 'Content-Type: application/json' -H 'NYT-App-Type: vhs' -H 'NYT-App-Version: v3.52.21' -H 'NYT-Token: MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAuNIzKBOFB77aT/jN/FQ+/QVKWq5V1ka1AYmCR9hstz1pGNPH5ajOU9gAqta0T89iPnhjwla+3oec/Z3kGjxbpv6miQXufHFq3u2RC6HyU458cLat5kVPSOQCe3VVB5NRpOlRuwKHqn0txfxnwSSj8mqzstR997d3gKB//RO9zE16y3PoWlDQXkASngNJEWvL19iob/xwAkfEWCjyRILWFY0JYX3AvLMSbq7wsqOCE5srJpo7rRU32zsByhsp1D5W9OYqqwDmflsgCEQy2vqTsJjrJohuNg+urMXNNZ7Y3naMoqttsGDrWVxtPBafKMI8pM2ReNZBbGQsQXRzQNo7+QIDAQAB' -H 'origin: https://cooking.nytimes.com' -H 'Referer: https://cooking.nytimes.com/' -H 'Connection: keep-alive' --data-raw '{"query":"\nquery VideoQuery($id: String!) {\n  video(id: $id) {\n    ...on Video {\n      advertisingProperties {\n        sensitivity\n        sponsored\n      }\n      bylines {\n        renderedRepresentation\n      }\n      contentSeries\n      cues {\n        name\n        type\n        timeIn\n        timeOut\n      }\n      duration\n      embedded\n      headline {\n        default\n      }\n      is360\n      isLive\n      liveUrls\n      playlist {\n        headline {\n          default\n        }\n        promotionalHeadline\n        url\n        sourceId\n        section {\n          displayName\n        }\n        videos(first: 20) {\n          edges @filterEmpty {\n            node {\n              advertisingProperties {\n                sensitivity\n                sponsored\n              }\n              id\n              sourceId\n              duration\n              section {\n                id\n                name\n              }\n              headline {\n                default\n              }\n              renditions {\n                url\n                type\n              }\n              url\n              promotionalMedia {\n                ... on Image {\n                  crops(cropNames: [SMALL_SQUARE, MEDIUM_SQUARE, SIXTEEN_BY_NINE]) {\n                    renditions {\n                      name\n                      width\n                      height\n                      url\n                    }\n                  }\n                }\n              }\n            }\n          }\n        }\n      }\n      promotionalHeadline\n      promotionalMedia {\n        ... on Image {\n          crops (cropNames: [SMALL_SQUARE, MEDIUM_SQUARE, SIXTEEN_BY_NINE, THREE_BY_TWO, TWO_BY_THREE, FLEXIBLE]) {\n            name\n            renditions {\n              name\n              width\n              height\n              url\n            }\n          }\n        }\n      }\n      promotionalSummary\n      related {\n        ... on Article {\n          promotionalHeadline\n          url\n          sourceId\n        }\n      }\n      renditions {\n        type\n        width\n        height\n        url\n        bitrate\n      }\n      section {\n        name\n      }\n      shortUrl\n      sourceId\n      subsection {\n        name\n      }\n      summary\n      timesTags {\n        __typename\n        displayName\n        isAdvertisingBrandSensitive\n        vernacular\n      }\n      url\n    }\n  }\n}","variables":{"id":"nyt://video/bc47bb72-497a-5160-819e-a3c2a7cb27c3"}}'
```
So we have to work out how the JS makes the UUID out of the media asset id, and also, if it's not a long-lived value, how to make the NYT token.

**Comment by SirElderling** [comment]
Thanks @dirkf for your response.
I can find the nyt-token embedded in pages like this: `https://www.nytimes.com/video/us/politics/100000009183577/biden-xi-apec.html`. The following items also appear: `"nyt-app-type": "project-vi"`, `"nyt-app-version": "0.0.5"`.

For the `https://cooking.nytimes.com/guides/13-how-to-cook-a-turkey` I could find the token hardcoded here: `webpack:///app/assets/web/js/properties.js`. 
In this case, the app-type and app-version are different from the example above: `"Nyt-App-Type": "vhs"`, `"Nyt-App-Version": "v3.52.21"`

**Comment by SirElderling** [comment]
I've progressed in the main articles extraction.
Consider the effort I already had with this issue, I'll take care of adjusting the extractor and submitting the required patch.

**Comment by SirElderling** [comment]
It's been a while since I gave you an update. 

I've finished coding everything I plan to include in the extraction process, like recipes, cooking guides, audio, video, video playlist, and articles. However, I'm facing some challenges with refactoring and cleaning up the code. What I might end up doing is releasing an initial pull request (PR) with only the `Article` and `Recipes` extraction to keep on par with the existing non working version. 
Later on, I'll submit additional PRs to add the other parts that aren't as urgent.
